### DISCUSSION TITLE
[Feature request] midwayjs/koa 支持 asyncLocalStorage 选项

### DISCUSSION THREAD

**Discussion by qwang1113** [midwayjs/midway#3906] [Ideas]
### The feature, motivation（功能、动机）

[koa](https://github.com/koajs/koa/blob/master/lib/application.js#L81) 已经支持了在异步操作中存储和获取请求级别的 context

通过这个操作, 可以在 midway 作用域之外 (比如某些 pureFunction 中) 获取到当前流入的请求上下文, 这对于一些需要追踪链路又不想全部引入 opentelementry 的时候非常有用

但 midwayjs/koa 中还无法传入开启这个选项, https://github.com/midwayjs/midway/blob/main/packages/web-koa/src/framework.ts#L76, 希望支持

**Comment by czy88840616** [comment]
参考：https://github.com/midwayjs/midway/blob/main/packages/tenant/src/tenantManager.ts

**Reply by waitingsong** [reply]
就是说可以实现在单例的类中直接获取到 Context 之类的请求级别的对象了？

**Reply by czy88840616** [reply]
可以，但是没开放避免滥用

**Reply by waitingsong** [reply]
这个功能不错。单例类里面要做 otel 追踪的确有点麻烦，要么显式传入 ctx， 要么直接传入请求级别的 traceService。这下好弄了～
